test/release/examples/benchmarks/shootout/knucleotide.chpl

```diff
@@ -22,16 +22,21 @@ proc main(args: [] string) {
       lineSize = 0,
       numRead = 0;
 
-  while stdinNoLock.readline(buff, lineSize) && !startsWithThree(buff) do
+  lineSize = stdinNoLock.readLine(buff);
+  while lineSize>0 && !startsWithThree(buff) {
     numRead += lineSize;
-
+    lineSize = stdinNoLock.readLine(buff);
+  }
   // Read in the rest of the file
   var dataDom = {1..fileLen-numRead},
       data: [dataDom] uint(8),
       idx = 1;
 
-  while stdinNoLock.readline(data, lineSize, idx) do
+  lineSize = stdinNoLock.readLine(data[idx..]);
+  while lineSize>0 {
     idx += lineSize - 1;
+    lineSize = stdinNoLock.readLine(data[idx..]);
+  }
```

Comment:
These similarly seem like they could be rewritten without the duplication (and may be worth it given that they're in the release directory?).  Unless I'm missing something...?

Reply:
In the case of `readStidedArr.chpl` we weren't using the return value from `readLine` so it would be feasible to write it without duplication. Here we do use the return value (`lineSize`) inside the loop so I don't think we can rewrite it without duplication unless chapel allows us to assign to a var and check its value at the same time.

Reply:
Oh, good point.  Could a `do...while` loop be used to eliminate the duplication?  E.g.,

```chapel
do {
  const lineSize = stdinNoLock.readLine(data[idx..]);
  idx += lineSize - 1;
} while (lineSize > 0);
```

I suppose the problem is that if we read nothing, we'd be adding `-1` to `idx`.  Ah, but maybe this is a case where we'd want to strip the newline (in which case I think lineSize would be the right value?)

Reply:
I think it would still be off by one when we encounter EOF because `lineSize` would be `0` in that case and we would subtract 1 unnecessarily?

Reply:
I don't think so.  At least, assuming that readLine returns 6 for "12345\n" when preserving the newline and 5 when stripping it, I'd expect us to be able to do:

```chapel
do {
  const lineSize = stdinNoLock.readLine(data[idx..], stripNewline=true);
  idx += lineSize;
} while (lineSize > 0);
```

(???).  And even if I'm wrong and we just have a global off-by-one, we could add a `-= 1` outside of the loop rather than duplicating the readline call.